libsrc/common/_printf.s

```diff
@@ -135,9 +144,9 @@ ReadInt:
         pha                             ; Save digit value
         lda     ptr1
         ldx     ptr1+1
-        asl     ptr1
+        asl     a
         rol     ptr1+1                  ; * 2
-        asl     ptr1
+        asl     a
```

Comment:
sure this is the right thing to do? the original code will add the shifted value below - after the change its the non shifted value - or am i missing something? :)

Reply:
I was quite sue you'd say something, because I had to think twice about this one, but yes:
```
        lda     ptr1
        ldx     ptr1+1
        asl     ptr1
        rol     ptr1+1                  ; * 2
        asl     ptr1
        rol     ptr1+1                  ; * 4, assume carry clear
        adc     ptr1
        sta     ptr1
```
Let's say you LDA with ptr1=2, so now A=2
ASL ptr1 ; ptr1=4, A=2
ASL ptr1 ; ptr1=8, A=2
adc ptr1 ; A = 8+2 = 10
sta ptr1 ; ptr1 = 10
Now instead you you LDA with ptr1=2, so now A=2
ASL A ; ptr1=2, A=4
ASL A ; ptr1=2, A=8
adc ptr1 ; A = 2+8 = 10
sta ptr1 ; ptr1 = 10